src/imp.rs

```diff
@@ -222,27 +222,10 @@ where
             return Err(Error::NotEnoughBalance);
         }
 
-        // This sorts the notes from least valuable to most valuable. It
-        // helps in the minimum gas spent algorithm, where the largest notes
-        // are "popped" first.
-        notes_and_values.sort_by(|(_, aval, _), (_, bval, _)| aval.cmp(bval));
+        let inputs = pick_notes(value, notes_and_values)
+            .ok_or(Error::NoteCombinationProblem)?;
 
-        let mut inputs = Vec::with_capacity(notes.len());
-
-        let mut accumulated_value = 0;
-        while accumulated_value < value {
-            // This unwrap is ok because at this point we can be sure there
-            // is enough value in the notes.
-            let (note, val, blinder) = notes_and_values.pop().unwrap();
-            accumulated_value += val;
-            inputs.push((note, val, blinder));
-        }
-
-        if inputs.len() > MAX_INPUT_NOTES {
-            return Err(Error::NoteCombinationProblem);
-        }
-
-        let change = accumulated_value - value;
+        let change = inputs.iter().map(|(_, val, _)| val).sum::<u64>() - value;
```

Comment:
I think you can just `map` as: `map(|v| v.1)` it's shorter and still clear what it does.